src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonArray.cs

```diff
@@ -48,6 +48,83 @@ public JsonArray(params JsonNode?[] items) : base()
             InitializeFromArray(items);
         }
 
+        internal override JsonNode InternalDeepClone()
+        {
+            if (_jsonElement.HasValue)
+            {
+                return new JsonArray(_jsonElement.Value.Clone(), Options);
+            }
+
+            var jsonArray = new JsonArray(Options);
```

Comment:
Given that we know the size of the array, isn't it possible to pre-size the target JsonArray?